src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/TokenSearchValue.cs

```diff
@@ -107,9 +107,9 @@ public bool Equals([AllowNull] ISearchValue other)
                 return false;
             }
 
-            return string.Equals(System, tokenSearchValueOther.System, StringComparison.OrdinalIgnoreCase) &&
-                string.Equals(Code, tokenSearchValueOther.Code, StringComparison.OrdinalIgnoreCase) &&
-                string.Equals(Text, tokenSearchValueOther.Text, StringComparison.OrdinalIgnoreCase);
+            return string.Equals(System, tokenSearchValueOther.System, StringComparison.Ordinal) &&
+                string.Equals(Code, tokenSearchValueOther.Code, StringComparison.Ordinal) &&
+                string.Equals(Text, tokenSearchValueOther.Text, StringComparison.Ordinal);
```

Comment:
Maybe this needs a comment for future us 😃 along the lines of "using string comparison to match database collation". Does this have any impact on Cosmos? Might be worth a quick check

Reply:
The new E2E test runs on both DBs. 
Yes, this bug is also in Cosmos. 
The search parameters before the change:
<img width="152" height="69" alt="image" src="https://github.com/user-attachments/assets/0a328bd4-f517-4f02-86dd-e61b464c8185" />

and after the change:
<img width="169" height="131" alt="image" src="https://github.com/user-attachments/assets/f3f34002-b2cc-4fc4-96ca-f9fd68beaa37" />